Detectors/Base/include/DetectorsBase/NameGen.h

```diff
@@ -0,0 +1,136 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_NAME_GENERATOR_H_
+#define ALICEO2_NAME_GENERATOR_H_
+
+#include "DetectorsCommonDataFormats/DetID.h"
+#include "SimConfig/ConfigurableParam.h"
+#include "SimConfig/ConfigurableParamHelper.h"
+#include <string_view>
+
+/// \file NameGen.h
+/// \brief Definition of the Names Generator class
+
+namespace o2
+{
+namespace base
+{
+// class for book-keeping of the names for output files and trees
+struct NameGen : public o2::conf::ConfigurableParamHelper<NameGen> {
+
```

Comment:
looks good to me overall. 

Not sure if "Gen" is better than just "Conf" (we are not really generating names). 
Also the keys in this parameter sometimes refer to branches sometimes to files. Should we make this part of the keyname?

Reply:
Ok for Conf. Will add File to filenames, while the names for branches or trees are rather generic, so I would not detalize them.